### DISCUSSION TITLE
Storybook interaction tests click issue

### DISCUSSION THREAD

**Discussion by pedrosousa13** [adobe/react-spectrum#6768] [Q&A]
We are using `usePress` and `useHover` for our buttons. With storybook interactions tests, I don't know how to trigger a `click()`.

Example code:

```
 <Button as={'button'} variant='primary' data-testid={'button'}>
    Trigger modal
</Button>
```

```
import { expect, screen, within,  userEvent} from '@storybook/test';
...
play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
	// Only pressing enter works
    // await userEvent.type(canvas.getByRole('button'), '{enter}');
    await userEvent.click(canvas.getByTestId('button'));

    const modalOpen = screen.queryAllByTestId('modal-test');

    await expect(modalOpen).toHaveLength(1);
    await expect(modalOpen[0]).toBeInTheDocument();
  },
```

**Comment by snowystinger** [comment]
Looks similar to our test of that functionality https://github.com/adobe/react-spectrum/blob/48af5db61da527f1f4ede38a9eb59bde5cdfc255/packages/%40react-spectrum/picker/stories/Picker.stories.tsx#L182

might be good to compare though and see what might be different

Do you use an animation when opening with click vs with keyboard?

**Comment by pedrosousa13** [comment]
going to close this as from what I can tell, it's the interaction between radix and click that doesn't work. We are triggering a modal with our button that never opens :)
